hypervisor/src/arch/emulator/mod.rs

```diff
@@ -6,19 +6,20 @@
 
 extern crate thiserror;
 
+use crate::arch::x86::ExceptionVector;
 use core::fmt::Debug;
 use std::fmt::{self, Display};
 use thiserror::Error;
 
 #[derive(Clone, Copy, Error, Debug)]
-pub struct Exception<T: Debug> {
-    vector: T,
+pub struct Exception {
+    vector: ExceptionVector,
```

Comment:
This makes the code x86 specific. The idea behind using generics was to make it arch agnostic.

Reply:
I realized that when I discovered the Aarch64 build failure. I think I will just abandon this approach for now.